packages/htmlbars-syntax/lib/utils.js

```diff
@@ -17,7 +17,14 @@ export function parseComponentBlockParams(element, program) {
     attrNames.push(element.attributes[i].name);
   }
 
-  var asIndex = attrNames.indexOf('as');
+  var asIndex = -1;
+
+  for (i = 0; i < attrNames.length; i++) {
+    if (attrNames[i] === 'as') {
+      asIndex = i;
+      break;
+    }
+  }
```

Comment:
Wow, I had no idea.  For the `for` loop, please use `i < l` rather than `i < attrNames.length`.  It's customary to do this for performance reasons, and you often see stuff like `for (var i=0, l=items.length; i < l; i++)` in the ember codebase.  In this case, we already have `l` from above.